File: spC++/src/Implement/Models/Table.cpp
#include "Headers/Models/Table.hpp"

Table::Table(std::string title) {
    this->title = title;
}

Table::Table(Table* table) {
    this->title = table->title;
    this->children = table->children;
}

IElement *Table::clone() {
    return new Table(this);
}

void Table::accept(IVisitor* v) {
    v->visitTable(this);
}

std::string Table::getTitle() {
    return this->title;
}

File: Java-Labs/lab/src/main/java/sp/lab/Models/Table.java
package sp.lab.Models;

import lombok.Getter;

public class Table extends Element{
    @Getter
    private String title;

    public Table(String title) {
        this.title = title;
    }

    public void accept(IVisitor visitor) {
        visitor.visitTable(this);
    }

    public void print() {
        System.out.println("Table with title: " + this.title);
    }
}
